.github/workflows/cmake.yml

```diff
@@ -77,3 +77,11 @@ jobs:
       working-directory: ${{github.workspace}}/build
       shell: bash
       run: ctest
+
+    - name: Test Stream List Replacement
```

Comment:
Yeah moving it to a new workflow is probably the best, this was in a way just proving that it could be done. I will move it and split the steps up.
Not sure why you would like it changed to "Statically Allocated"  there is no static allocations in the test. The test is simply to prove that it is possible to replace the internal stream list implementation with a different one by using the SRTP_NO_STREAM_LIST define.